File: python/day-01.py
#  Problem :  Given an array of size N, process each element exactly once.and found the lenght to array(list)
my_Array=[1,"Mani",18,20,'K',444] # python array is store to dfrent values not Homogeneous
my_Array2=[2,5,6,1,6,20,7,9,10,13,0,0.5,0.1,-1,-20] # Homogeneous Array
len=0
for elament in my_Array:
    print(f"Procesin the element :{elament}")
    len+=1
print(f"Lenth os the Array is:{len}")
# max of Array
max=my_Array2[0] # init to max is first elements
for item in my_Array2:
    if max< item:
        max=item
print(f"Max of Array is :{max}")
# min of the Array 
min=max
for item in my_Array2:
    if min > item:
        min=item
print(f"Min of Array is :{min}")
exit()

File: cpp/day-01.cpp
#include <iostream>
using namespace std;

int main() {

    // Mixed type → stored as string (C++ arrays must be same type)
    string my_Array[] = {"1","Mani","18","20","K","444"};

    // Homogeneous numeric array
    double my_Array2[] = {2,5,6,1,6,20,7,9,10,13,0,0.5,0.1,-1,-20};
    int len = 0;
    for(string element : my_Array) {
        cout << "Processing the element : " << element << endl;
        len++;
    }
    cout << "Length of the Array is : " << len << endl;
// Max of the Array
    double maxVal = my_Array2[0];   // initialize with first element

    for(double item : my_Array2) {
        if(maxVal < item) {
            maxVal = item;
        }
    }
    cout << "Max of Array is : " << maxVal << endl;
// Min of the Array
    double minVal = my_Array2[0];   // initialize with first element
    for(double item : my_Array2) {
        if(minVal > item) {
            minVal = item;
        }
    }
    cout << "Min of Array is : " << minVal << endl;
    return 0;
}
